### DISCUSSION TITLE
Ricardian contracts and triple entry accounting

### DISCUSSION THREAD

**Discussion by dilgreen** [CredComSoc/ProtocolWorkingGroup#12] [General]
Ian Grigg talked about both, very lucidly.
We should have an opinion on both, I think.

https://youtu.be/zjSd6wswc9c

**Comment by JohnG-cc** [comment]
If I understand correctly, “Ricardian Contract” here refers to a particular document format developed by Grigg, that contains machine readable metadata and is signed with a cryptographic hash. It also specifies that a transaction system using these contacts should require users to become party to the contract when transacting, requiring cryptographic signing of transactions to explicitly indicate acceptance of the contract.

If I also understand correctly, Triple Entry Accounting here refers not to the “Trebits” proposed in 1986 by the ex-president of the American Accounting Association, professor Yuji Iriji, but rather to the co-option (without attribution) of Iriji’s term by Grigg in 2005 to refer to digitally signed transaction records held on a single public ledger, rather than by a third party notary as proposed by Iriji.

If I understand the thesis of this presentation correctly, it is that the contracts underlying a blockchain and a user’s interaction with it should be made explicit, and by extension, blockchain systems cannot and should not be anonymous and should operate within existing legal frameworks.

Taking all of these together, I believe the relevant points are:

Transactions should be digitally signed: this is something that I have discussed with @utunga and mentioned here as well. I think CredCom needs this.

Third-party or public transaction records are necessary: I think this misses the point of digital signatures for non-repudiation. If you have someone’s signed receipt, you don’t need a notary in order to prove the transaction occurred. “Triple-Entry Accounting” is a misnomer in Grigg’s usage of the term, and I suspect it’s enthusiastic embrace by the blockchain community is a symptom of blockchain being a solution looking for problems. For CredCom, I don’t see any need for third party notaries or public ledgers as long as transactions are cryptographically signed and ownership of the keys used for signing can be clearly established (perhaps even by third party notary?)

Contracts should be explicit: I had assumed that CredCom would come into existence through contracts between members and a legal entity for their local trading group, and between the legal entities of various trading groups, that laid out the rules, rights, and obligations of the system. We certainly need something like that. I think Grigg’s Ricardian Contract is very much a creature of permissionless public blockchains, and as such cannot avail itself of member-joining-time contracts, and so is forced to wrestle with the complexity of transaction-time contracts. I see no place for the specific Ricardian Contract document format or transaction protocol in CredCom, which is naturally suited to simple and traditional membership contracts.
